import/xsl/subjectsplus.xsl

```diff
@@ -0,0 +1,183 @@
+<!-- available fields are defined in solr/biblio/conf/schema.xml -->
+<xsl:stylesheet version="1.0"
+    xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
+    xmlns:oai_dc="http://www.openarchives.org/OAI/2.0/oai_dc/"
+    xmlns:dc="http://purl.org/dc/elements/1.1/"
+    xmlns:php="http://php.net/xsl"
+    xmlns:xlink="http://www.w3.org/2001/XMLSchema-instance">
+    <xsl:output method="xml" indent="yes" encoding="utf-8"/>
+    <xsl:param name="institution">My University</xsl:param>
+    <xsl:param name="collection">SubjectsPlus</xsl:param>
+    <xsl:param name="id_tag_name">identifier</xsl:param>
+    <xsl:param name="change_tracking_core">biblio</xsl:param>
+    <xsl:param name="change_tracking_date_tag_name"></xsl:param>
+    <xsl:param name="workKey_include_regEx"></xsl:param>
+    <xsl:param name="workKey_exclude_regEx"></xsl:param>
+    <xsl:param name="workKey_transliterator_rules">:: NFD; :: lower; :: Latin; :: [^[:letter:] [:number:]] Remove; :: NFKC;</xsl:param>
+    <xsl:template match="/">
+        <xsl:if test="collection">
+            <collection>
+            <xsl:for-each select="collection">
+                <xsl:for-each select="oai_dc:dc">
+                    <xsl:apply-templates select="."/>
+                </xsl:for-each>
+            </xsl:for-each>
+            </collection>
+        </xsl:if>
+        <xsl:if test="oai_dc:dc">
+            <xsl:apply-templates/>
+        </xsl:if>
+    </xsl:template>
+    <xsl:template match="oai_dc:dc">
+        <add>
+            <doc>
+                <!-- ID -->
+                <!-- Important: This relies on an <identifier> tag being injected by the OAI-PMH harvester. -->
+                <field name="id">
+                    <xsl:value-of select="*[name()=$id_tag_name]"/>
+                </field>
+
+                <!-- RECORD FORMAT -->
+                <field name="record_format">subjectsplus</field>
+
+                <!-- FULLRECORD -->
+                <!-- disabled for now; records are so large that they cause memory problems!
+                <field name="fullrecord">
+                    <xsl:copy-of select="php:function('VuFind::xmlAsText', .)"/>
+                </field>
+                  -->
+
+                <!-- ALLFIELDS -->
+                <field name="allfields">
+                    <xsl:value-of select="normalize-space(string(.))"/>
+                </field>
+
+                <!-- INSTITUTION -->
+                <field name="institution">
+                    <xsl:value-of select="$institution" />
+                </field>
+
+                <!-- COLLECTION -->
+                <field name="collection">
+                    <xsl:value-of select="$collection" />
+                </field>
+
+                <!-- LANGUAGE -->
+                <xsl:if test="dc:language">
```

Comment:
I'd leave out the `xsl:if` surrounding `xsl:for-each` in each such case.

Reply:
I've taken care of this. I didn't mess with the top-level if statements that control the multi-part vs. single-part indexing. I suspect that can be simplified too, but it somehow felt clearer to leave the parallel if's in place.